backend/models/transaction.py

```
@@ -80,14 +181,40 @@ async def get_transactions_by_email(user_email: str = None, limit: int = None):
+                
+            # Validate status transition
+            if not is_valid_status_transition(current_tx.status, new_status):
+                logging.warning(f"Invalid status transition for {transaction_id}: {current_tx.status} -> {new_status}")
+                # Allow it but log the warning for now during migration
+                
+            result = await DB.update_document("transactions", {"transaction_id": transaction_id}, {"status": new_status})
+            
+            if result > 0:
+                logging.info(f"Transaction {transaction_id} status updated: {current_tx.status} -> {new_status}")
+            
+            return result > 0
```

Ruff (0.14.2)

Consider moving this statement to an `else` block

(TRY300)